app/index.js

```diff
@@ -64,56 +98,56 @@ app.on('ready', () => {
-            message: alertMessage,
-            buttons: ['No', 'Yes'],
-            cancelId: 0,
-            defaultId: 1
-          })
-          if (btnId === 1) {
-            // Use running node, skip starting the daemon
-            // Set repository path to null so it's ignored and not passed as IPFS_PATH
-            // Set binary path to `ipfs` to use the client's binary
-            skipRepoPath()
-            setCustomBinaryPath('ipfs')
-            return Promise.resolve()
+          console.log('Another service on localhost:5001 has been deteced')
+          const useExistingNode = askWhichNodeToUse(apiVersion)
+
+          if (useExistingNode) {
+            console.log('Using existing IPFS node (localhost:5001)')
+            global.IPFS_BINARY_PATH = 'ipfs'
+            global.IPFS_REPO_PATH = ''
+            return Promise.resolve(false) // it should not start the ipfs daemon
           } else {
             // Use our own daemon, but on different ports
-            customPorts = true
+            console.log('Using custom setup for Orion new IPFS node (localhost:5101)')
+            global.IPFS_MULTIADDR_API = '/ip4/127.0.0.1/tcp/5101'
+            global.IPFS_MULTIADDR_GATEAY = '/ip4/127.0.0.1/tcp/8180'
+            global.IPFS_MULTIADDR_SWARM = `'["/ip4/0.0.0.0/tcp/4101", "/ip6/::/tcp/4101"]'`
           }
         }
-
+        return Promise.resolve(true) // it should start the ipfs daemon
+      })
+      .then((shouldStart) => {
+        // Logs the path and configuration used
+        console.log('IPFS_BINARY_PATH', global.IPFS_BINARY_PATH)
+        console.log('IPFS_MULTIADDR_API', global.IPFS_MULTIADDR_API)
+        console.log('IPFS_MULTIADDR_GATEAY', global.IPFS_MULTIADDR_GATEAY)
+        console.log('IPFS_MULTIADDR_SWARM', global.IPFS_MULTIADDR_SWARM)
+        console.log('IPFS_REPO_PATH', global.IPFS_REPO_PATH)
+        console.log('IPFS_MULTIADDR_API', global.IPFS_MULTIADDR_API)
+        return Promise.resolve(shouldStart)
+      })
+      .then((shouldStart) => {
+        if (!shouldStart) return Promise.resolve()
+        // Starts the IPFS daemon
+        console.log('IPFS Daemon: Starting')
         return ensuresIPFSInitialised()
-          .then(() => ensureAddressesConfigured(customPorts))
+          .then(() => ensureDaemonConfigured())
           .then(startIPFSDaemon)
           .then((process) => {
-            console.log('IPFS Daemon: Starting')
             global.IPFS_PROCESS = process
             loadingWindow.webContents.send('set-progress', {
               text: 'Initializing the IPFS Daemon...',
               percentage: 20
             })
             return Promise.resolve()
           })
-          // Wait for the repo to be unlocked
-          // (usually ipfs daemon needs some time before it releases the lock)
-          .then(promiseRepoUnlocked)
       })
+      .then(promiseRepoUnlocked) // ensures that the api are ready
+      .then(() => ensureDaemonConfigured())
```

Comment:
`ensureDaemonConfigured` is already called on `136`, do we need it again?

Reply:
One is for the daemon that starts, the other one is to ensure that the configuration is correct for the already running one.